app/code/Magento/ConfigurableProduct/Plugin/Model/ResourceModel/Product.php

```diff
@@ -51,6 +57,29 @@ public function beforeSave(
         /** @var \Magento\Catalog\Model\Product $object */
         if ($object->getTypeId() == Configurable::TYPE_CODE) {
             $object->getTypeInstance()->getSetAttributes($object);
+            $this->resetConfigurableOptionsData($object);
+        }
+    }
+
+    /**
+     * Set null for configurable options attribute of configurable product
+     *
+     * @param \Magento\Catalog\Model\Product $object
+     * @return void
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function resetConfigurableOptionsData($object)
+    {
+        $extensionAttribute = $object->getExtensionAttributes();
+        if ($extensionAttribute && $extensionAttribute->getConfigurableProductOptions()) {
+            /** @var ProductAttributeRepositoryInterface $productAttributeRepository */
+            $productAttributeRepository = \Magento\Framework\App\ObjectManager::getInstance()
+                ->get(ProductAttributeRepositoryInterface::class);
+            /** @var OptionInterface $option */
+            foreach ($extensionAttribute->getConfigurableProductOptions() as $option) {
+                $eavAttribute = $productAttributeRepository->get($option->getAttributeId());
```

Comment:
Loading attributes in the loop isn't best idea

Reply:
Hi @kandy 
Could you suggest me an alternative way?

Reply:
I've updated it